Automattic/wp-calypso
# Stats notices: Refactoring for better readability

Related to #80951

## Proposed Changes

The PR proposes to calculate the visibility of notices in the same place, so that we could ensure we show only one notice at the same time.

The PR proposes the following changes:

- Refactored all notice settings to `all-notice-definitions.ts`, including the post purchase notices and the upgrade notice
  - Introduces `ALL_STATS_NOTICES` which is an array of stats settings
  - Introduces a `isVisibleFunc` for each definition, which is used to determine whether we should show the notice or not
- Refactored  `use-notice-visibility-query.ts`
  - Introduces `useNoticesVisibilityQueryRaw` for the retrieval of all notice visibility
  - Introduces default visibility for server and client notice
  - Client notices all have the default value of `true`

## Testing Instructions

* Open Calypso Live Branch
* Open `/stats/day/:siteSlug` for a site without Stats products
* Ensure you see the upgrade notice
* Add `?statsPurchaseSuccess=free` to the URL
* Ensure you only see the upgrade success notice for the Free product
* Add `?statsPurchaseSuccess=paid` to the URL
* Ensure you only see the upgrade success notice for the Paid products

Repeat the same for Odyssey Stats, and ensure the notices work the same way as they are on Calypso.

## Pre-merge Checklist
